### PR TITLE
Feature product endpoint

### PR DESCRIPTION
**Pull request by RRicardotj**
This PR will create
Product seeders
Product query schema
products controller

### CODE REVIEW

File: src/controllers/product/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { product } from '~/src/models'
+
+export const products = () => product.findAll()
```
Comment (abelosorio) [inline src/controllers/product/index.js]: Functions should be named after verbs.

File: src/schema/types/product.graphql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  id: ID!
+  photoUrl: String!
+  name: String!
+  description: String
+  price: Float!
+  marketPrice: Float!
+}
+
+extend type Query {
+  products: [Product]!
```
Comment (abelosorio) [inline src/schema/types/product.graphql]: This should be `[Products!]!`.

File: src/controllers/product/index.js (hunk 1 above)
Comment (RRicardotj) [reply src/controllers/product/index.js]: can be a verb like: getProducts?

File: src/schema/types/product.graphql (hunk 2 above)
Comment (RRicardotj) [reply src/schema/types/product.graphql]: ummm ok, the product array can not be empty

Comment (abelosorio) [reply src/schema/types/product.graphql]: It actually can. The double bang requires the controller to return either an array or nothing. With only one (`[]!`) the controller could return an array of empty values in it: `[product1, product2, null, undefined, false]`.

File: src/controllers/product/index.js (hunk 1 above)
Comment (abelosorio) [reply src/controllers/product/index.js]: That's right.

File: src/schema/types/product.graphql (hunk 2 above)
Comment (RRicardotj) [reply src/schema/types/product.graphql]: Aaah! Got it! Thanks @abelosorio

File: src/controllers/product/index.js (hunk 1 above)
Comment (RRicardotj) [reply src/controllers/product/index.js]: Fixed!, also I've created an Internal error type to handle errors that client should not know. Please check it out

File: src/controllers/product/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import InternalError from '~/src/errors/internal-error'
+import { product } from '~/src/models'
+
+export const getProducts = () => product.findAll()
+  .catch(() => {
```
Comment (abelosorio) [inline src/controllers/product/index.js]: I'd suggest catching errors at a higher level so that we keep our controllers lean.

Comment (RRicardotj) [reply src/controllers/product/index.js]: I've created a handler for this, please give me your point of view of that

**Pull request merged**